### PR TITLE
15531 Legal name updates for email reminder job

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#15531

*Description of changes:*
- Updated project to use `pyproject.toml`
- Changed `Business` model references to `LegalEntity`, and other related updates
- Updated to publish message to GCP emailer queue
- Updated Dockerfile

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/email-reminder/k8s/knative/dc.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        - name: DATABASE_USERNAME
+          value: DATABASE_USERNAME
+        - name: AUDIENCE
+          value: $AUDIENCE
+        - name: PUBLISHER_AUDIENCE
+          value: $PUBLISHER_AUDIENCE
+        - name: ENTITY_MAILER_TOPIC
+          value: $ENTITY_MAILER_TOPIC
+        - name: ENTITY_FILER_TOPIC
+          value: $ENTITY_FILER_TOPIC
```
Comment (argush3) [inline jobs/email-reminder/k8s/knative/dc.yaml:51]: don't think we need the entity filer topic for this job

File: jobs/email-reminder/k8s/knative/dc.env-template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+DATABASE_PASSWORD=
+DATABASE_PORT=
+GUNICORN_PROCESSES=
+GUNICORN_THREADS=
+DATABASE_UNIX_SOCKET=
+DATABASE_USERNAME=
+AUDIENCE=
+PUBLISHER_AUDIENCE=
+ENTITY_MAILER_TOPIC=
+ENTITY_FILER_TOPIC=
```
Comment (argush3) [inline jobs/email-reminder/k8s/knative/dc.env-template:21]: filer topic can be removed

File: jobs/email-reminder/email_reminder.py [hunk 3]
```diff
@@ -19,16 +19,17 @@
 
 import requests
 import sentry_sdk  # noqa: I001, E501; pylint: disable=ungrouped-imports; conflicts with Flake8
-from flask import Flask
-from legal_api.models import Business, Filing, db  # noqa: I001
+from flask import Flask, current_app
+from legal_api.models import Filing, LegalEntity, db  # noqa: I001
```
Comment (thorwolpert) [inline jobs/email-reminder/email_reminder.py:23]: this should move to use the business-model
https://github.com/bcgov/lear/tree/dev_legal_name_changes/python/common/business-registry-model

**Review by thorwolpert — COMMENTED** [review]
migrate to just using the model

File: jobs/email-reminder/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+flake8-blind-except = "^0.2.1"
+flake8-debugger = "^4.1.2"
+flake8-docstrings = "^1.7.0"
+flake8-isort = "^6.0.0"
+flake8-quotes = "^3.3.2"
+pep8-naming = "^0.13.3"
+autopep8 = "^2.0.2"
+coverage = "^7.3.0"
+pylint = "^2.17.5"
+pylint-flask = "^0.6"
```
Comment (argush3) [inline jobs/email-reminder/pyproject.toml:56]: I think these can be removed.  

And please add `black` as a dev dependency

File: jobs/email-reminder/email_reminder.py (hunk 3 above)
Comment (argush3) [reply jobs/email-reminder/email_reminder.py:23]: @thorwolpert what do you want to do about this here? the email reminder job references some of the service classes in legal api - grabbing auth token & some feature flag stuff

Do you want us to cut another ticket to create a common module for services and come back and integrate? or just stick with importing all of the entire legal api module?

Comment (argush3) [reply jobs/email-reminder/email_reminder.py:23]: Approving PR to reference legal-api as a dependency @thorwolpert.

If you want any changes with regards to my previous comment, let me know.  I can cut a new ticket to address it.

**Review by argush3 — APPROVED** [review]

**Pull request merged**